test/common/json/json_loader_test.cc

```diff
@@ -445,5 +445,27 @@ TEST(JsonLoaderTest, YamlObject) {
   }
 }
 
+TEST(JsonLoaderTest, BadYamlException) {
+  std::string bad_yaml = R"EOF(
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: {{ ntop_ip_loopback_address }}
+      port_value: 0
+)EOF";
+
+  bool exception_caught = false;
+  std::string exception_text;
+  try {
+    const Json::ObjectSharedPtr json = Json::Factory::loadFromYamlString(bad_yaml);
+  } catch (const EnvoyException& e) {
+    exception_caught = true;
+    exception_text = e.what();
+  }
+  EXPECT_TRUE(exception_caught);
+  EXPECT_TRUE(exception_text.find("bad conversion") != std::string::npos) << exception_text;
```

Comment:
`EXPECT_THROW_WITH_REGEX`?

Reply:
I didn't find doc for that...but in any case I think that would probably be an adequate replacement for what I have now, but maybe not an adequate replacement for the negative test I just added, that shows we have explicitly looked for a BadConversion problem.

Reply:
It doesn't do *all* I want.  And I see what I couldn't find doc for it; it's defined in the Envoy codebase, so I added a parallel macro EXPECT_THROW_WITHOUT_REGEX and now I'm all set.

Having some trouble with docker for formatting fixups, and will push a new commit when that's sorted.